### ISSUE TITLE
UnnecessarySemicolonAfterTypeMemberDeclarationCheck: trailing semicolon after nested type not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20271]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/coding/unnecessarysemicolonaftertypememberdeclaration.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputUnnecessarySemicolonAfterTypeMemberDeclarationCompact.java <<'EOF'
void main() { }

class Inner { };
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputUnnecessarySemicolonAfterTypeMemberDeclarationCompact.java

Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

UnnecessarySemicolonAfterTypeMemberDeclarationCheck must flag the trailing `;` after `class Inner { }` in this compact file. By JEP 512 semantics `Inner` is a member of the implicit class that wraps the file, exactly analogous to a nested class inside an ordinary outer class. The trailing semicolon after a member-type declaration is what this check is for.

Expected on the compact input: one violation `Unnecessary semicolon.` on line 3.

Root cause: the check guards on `parent == OBJBLOCK` to recognise type-member context. Top-level type declarations in a compact source file have parent `COMPACT_COMPILATION_UNIT` instead, so the check does not see them as member declarations and the trailing semicolon goes unflagged.

This pairs with the false-positive case where `UnnecessarySemicolonAfterOuterTypeDeclaration` fires on the same site.

**Same code inside an ordinary class:**

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > InputUnnecessarySemicolonAfterTypeMemberDeclarationOrdinary.java <<'EOF'
public class InputUnnecessarySemicolonAfterTypeMemberDeclarationOrdinary {
    public void main() { }

    class Inner { };
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.6.0-SNAPSHOT-all.jar -c config.xml InputUnnecessarySemicolonAfterTypeMemberDeclarationOrdinary.java

Starting audit...
[ERROR] InputUnnecessarySemicolonAfterTypeMemberDeclarationOrdinary.java:4:20: Unnecessary semicolon. [UnnecessarySemicolonAfterTypeMemberDeclaration]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
fix is merged
